docs/swift/interceptors.md

```diff
@@ -6,40 +6,85 @@ sidebar_position: 5
 Interceptors are a powerful way to observe and mutate outbound and inbound
 headers, data, trailers, and errors both for unary APIs and streams.
 
-An interceptor is instantiated **once for each request** and provides a set of
-closures that are invoked by the client during the lifecycle of that request.
-Each closure provides the ability for the interceptor to observe and store
-state, as well as the option to mutate the outbound or inbound content.
+Each interceptor is instantiated **once per request or stream** and
+provides a set of closures that are invoked by the client during the lifecycle
+of that call. Each closure allows the interceptor to observe and store
+state, as well as to mutate outbound or inbound content.
+
+Every interceptor has the opportunity to perform asynchronous work before passing a potentially
+altered value to the next interceptor in the chain. When the end of the chain is reached, the
+final value is passed to the networking client where it is sent to the server (outbound request)
+or to the caller (inbound response).
+
+Interceptors may also fail outbound requests before they're sent, thus preventing subsequent
+interceptors from being invoked and returning a specified error back to the original caller.
+
+Interceptors are closure-based and are passed both the current value and a closure which
+should be called to resume the interceptor chain. Propagation will not continue until
+this closure is called. Additional values may still be passed to a given interceptor even
+though it has not yet continued the chain with a previous value. For example:
+
+- A request is sent
+- Response headers are received, and an interceptor pauses the chain while processing them
+- First chunk of streamed data is received, and the interceptor is invoked with this value
+- Interceptor is expected to resume with headers first, and then with data after
```

Comment:
Possibly better as an ordered list?

Reply:
As in - in the documentation, or resuming in code by passing a list?

Reply:
I think @akshayjshah is referring to the docs here. Since those steps happen in sequence, it may make sense to represent them in an ordered list (1, 2, 3...).

Reply:
👍🏽 updated